### PR TITLE
fix: last deployed stack local file is overwritten when different apps are used

### PR DESCRIPTION
**Pull request by philasmar**
*Issue #, if available:*
IDE-6121

*Description of changes:*
fix: last deployed stack local file is overwritten when different apps are used

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/AWS.Deploy.Orchestration/LocalUserSettings/LocalUserSettingsEngine.cs [hunk 1]
```diff
@@ -67,12 +67,23 @@ public async Task UpdateLastDeployedStack(string stackName, string projectName,
                     }
                     else
                     {
-                        localUserSettings.LastDeployedStacks = new List<LastDeployedStack>() {
-                            new LastDeployedStack(
-                                awsAccountId,
-                                awsRegion,
-                                projectName,
-                                new List<string>() { stackName })};
+                        var currentStack = new LastDeployedStack(
```
Comment (normj) [inline src/AWS.Deploy.Orchestration/LocalUserSettings/LocalUserSettingsEngine.cs:70]: This object should be created inside the if block that is actually using it.

File: src/AWS.Deploy.Orchestration/LocalUserSettings/LocalUserSettingsEngine.cs [hunk 2]
```diff
@@ -67,12 +67,23 @@ public async Task UpdateLastDeployedStack(string stackName, string projectName,
...
+                            awsRegion,
+                            projectName,
+                            new List<string>()
+                            {
+                                stackName
+                            });
+
+                        if (localUserSettings.LastDeployedStacks == null)
+                        {
+                            localUserSettings.LastDeployedStacks = new List<LastDeployedStack>() { currentStack };
```
Comment (ganeshnj) [inline src/AWS.Deploy.Orchestration/LocalUserSettings/LocalUserSettingsEngine.cs:81]: is it worth changing `LastDeployedStacks` to non-nullable with constructor initialization or with default empty collection?

Comment (philasmar) [reply src/AWS.Deploy.Orchestration/LocalUserSettings/LocalUserSettingsEngine.cs:81]: Done

**Comment by philasmar** [comment]
> Is this possible to add a unit test using mocks?

Done

File: src/AWS.Deploy.Orchestration/LocalUserSettings/LocalUserSettingsEngine.cs (hunk 2 above)
Comment (ganeshnj) [reply src/AWS.Deploy.Orchestration/LocalUserSettings/LocalUserSettingsEngine.cs:81]: you can remove the whole null check now

**Pull request closed without merging**